### PR TITLE
#3600 Add sharding-scaling-distribution module

### PR DESCRIPTION
**Pull request by ssxlulu**
#3600 Add sharding-scaling-distribution module

### CODE REVIEW

File: sharding-distribution/sharding-scaling-distribution/src/main/resources/conf/log4j.properties [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+log4j.rootLogger=DEBUG, root
+log4j.appender.root=org.apache.log4j.ConsoleAppender
+log4j.appender.root.layout=org.apache.log4j.PatternLayout
+log4j.appender.root.layout.ConversionPattern=%d [%t] %p %l - %m%n
```
Comment (dongzl) [inline sharding-distribution/sharding-scaling-distribution/src/main/resources/conf/log4j.properties]: I understand ShardingSphere use logback as log component, do we need log4j?

Comment (ssxlulu) [reply sharding-distribution/sharding-scaling-distribution/src/main/resources/conf/log4j.properties]: Yes, we should transform the log4j to logback, a new issue may be submitted.

Comment (KomachiSion) [reply sharding-distribution/sharding-scaling-distribution/src/main/resources/conf/log4j.properties]: I think we need to use logback replace log4j, we can change in other PR.
This Pr for adding sharding-scaling-distribution module only, What do you think about?

Comment (dongzl) [reply sharding-distribution/sharding-scaling-distribution/src/main/resources/conf/log4j.properties]: OK, I understand.

Comment (ssxlulu) [reply sharding-distribution/sharding-scaling-distribution/src/main/resources/conf/log4j.properties]: Yes, I agree with you @KomachiSion .

Comment (KomachiSion) [reply sharding-distribution/sharding-scaling-distribution/src/main/resources/conf/log4j.properties]: @dongzl , If you think the PR can be merged, can you approve it?

File: sharding-distribution/sharding-scaling-distribution/src/main/assembly/sharding-scaling-binary-distribution.xml [hunk 2]
```diff
@@ -51,4 +47,11 @@
...
         </fileSet>
     </fileSets>
+
+    <dependencySets>
+        <dependencySet>
+            <outputDirectory>lib</outputDirectory>
+            <fileMode>0644</fileMode>
+        </dependencySet>
+    </dependencySets>
 </assembly>
```
Comment (dongzl) [inline sharding-distribution/sharding-scaling-distribution/src/main/assembly/sharding-scaling-binary-distribution.xml]: need a new blank line at end of file.

File: sharding-distribution/sharding-scaling-distribution/src/main/resources/conf/config.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        "url": "jdbc:mysql://127.0.0.1:3306/test2?serverTimezone=UTC&useSSL=false",
+        "username": "root"
+      }
+  },
+  "serverConfiguration": {
+    "blockQueueSize": 10000,
+    "pushTimeout": 1000,
+    "concurrency": 3
+  }
+}
```
Comment (dongzl) [inline sharding-distribution/sharding-scaling-distribution/src/main/resources/conf/config.json]: need a new blank line at end of file.

File: sharding-scaling/sharding-scaling-bootstrap/pom.xml [hunk 4]
```diff
@@ -66,27 +66,4 @@
...
-                        <phase>package</phase>
-                        <goals>
-                            <goal>single</goal>
-                        </goals>
-                    </execution>
-                </executions>
-            </plugin>
-        </plugins>
-    </build>
 </project>
```
Comment (dongzl) [inline sharding-scaling/sharding-scaling-bootstrap/pom.xml]: need a new blank line at end of file.

**Comment by dongzl** [comment]
hi @ssxlulu , This is a conflicting files, should deal with it.

**Comment by ssxlulu** [comment]
OK.

**Comment by ssxlulu** [comment]
> @ssxlulu the PR has conflict, Can you solve it first?

Ok.

**Comment by KomachiSion** [comment]
@ssxlulu , CI failed, Can you check and fix it?

**Comment by ssxlulu** [comment]
> @ssxlulu , CI failed, Can you check and fix it?

The CI shows success.

image

**Pull request merged**